tests/phpunit/tests/comment/query.php

```diff
@@ -5372,4 +5383,175 @@ public function test_query_does_not_have_leading_whitespace() {
 
 		$this->assertSame( ltrim( $q->request ), $q->request, 'The query has leading whitespace' );
 	}
+
+	/**
+	 * Helper method to create standard test comments for note type exclusion tests.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @return int[] Array of comment IDs indexed by type: [0] => regular comment, [1] => pingback, [2] => note.
+	 */
+	protected function create_note_type_test_comments() {
+		$comments    = array();
+		$comments[0] = self::factory()->comment->create(
+			array(
+				'comment_post_ID'  => self::$post_id,
+				'comment_approved' => '1',
+			)
+		);
+		$comments[1] = self::factory()->comment->create(
+			array(
+				'comment_post_ID'  => self::$post_id,
+				'comment_approved' => '1',
+				'comment_type'     => 'pingback',
+			)
+		);
+		$comments[2] = self::factory()->comment->create(
+			array(
+				'comment_post_ID'  => self::$post_id,
+				'comment_approved' => '1',
+				'comment_type'     => 'note',
+			)
+		);
```

Comment:
Maybe better?

```suggestion
		$comments = array(
			self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
				)
			),
			self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
					'comment_type'     => 'pingback',
				)
			),
			self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
					'comment_type'     => 'note',
				)
			)
		);
```

Reply:
Alternatively, with other refactoring:

```suggestion
		$comments = array(
			'comment' => self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
				)
			),
			'pingback' => self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
					'comment_type'     => 'pingback',
				)
			),
			'note' => self::factory()->comment->create(
				array(
					'comment_post_ID'  => self::$post_id,
					'comment_approved' => '1',
					'comment_type'     => 'note',
				)
			)
		);
```